core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -274,10 +291,12 @@ public static TableMetadata read(FileIO io, String path) {
   }
 
   public static TableMetadata read(FileIO io, InputFile file) {
+    EncryptionManager encryption =
+        (io instanceof EncryptingFileIO) ? ((EncryptingFileIO) io).encryptionManager() : null;
```

Comment:
There should be no need for most of the changes in this file after my recommendations on #7770.